sql_dialect.go

```diff
@@ -37,18 +38,23 @@ type (
 var (
 	dialects              = make(map[string]SQLDialect)
 	DefaultDialectOptions = sqlgen.DefaultDialectOptions
+	dialectsMu            sync.RWMutex
 )
 
 func init() {
 	RegisterDialect("default", DefaultDialectOptions())
 }
 
 func RegisterDialect(name string, do *SQLDialectOptions) {
+	dialectsMu.Lock()
```

Comment:
similar to this: https://github.com/golang/go/blob/master/src/database/sql/sql.go#L45
faced problem with concurrent map writes in our test environment